modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,703 @@
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to remote_process for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+  
+  RpTools <- reticulate::import("RpTools")
+  
+  input_file <- NULL
+  stage_get_data <- NULL
+  stage_process_data <- NULL
+  raw_merge <- NULL
+  pro_merge <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
+  
+  # extract the variables from the settings list
+  siteid <- as.numeric(settings$run$site$id)
+  siteid_short <- paste0(siteid %/% 1e+09, "-", siteid %% 1e+09)
+  raw_mimetype <- settings$remotedata$raw_mimetype
+  raw_formatname <- settings$remotedata$raw_formatname
+  outdir <- settings$outdir
+  start <- as.character(as.Date(settings$run$start.date))
+  end <- as.character(as.Date(settings$run$end.date))
+  source <- settings$remotedata$source
+  collection <- settings$remotedata$collection
+  scale <- settings$remotedata$scale
+  if (!is.null(scale)) {
+    scale <- as.double(settings$remotedata$scale)
+    scale <- format(scale, nsmall = 1)
+  }
+  projection <- settings$remotedata$projection
+  qc <- settings$remotedata$qc
+  if (!is.null(qc)) {
+    qc <- as.double(settings$remotedata$qc)
+    qc <- format(qc, nsmall = 1)
+  }
+  algorithm <- settings$remotedata$algorithm
+  credfile <- settings$remotedata$credfile
+  pro_mimetype <- settings$remotedata$pro_mimetype
+  pro_formatname <- settings$remotedata$pro_formatname
+  out_get_data <- settings$remotedata$out_get_data
+  out_process_data <- settings$remotedata$out_process_data
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  on.exit(PEcAn.DB::db.close(dbcon), add = TRUE)
+  
+  flag <- 0
+  
+  # collection dataframe used to map Google Earth Engine collection names to their PEcAn specific names
+  collection_lut <- data.frame(
+    stringsAsFactors = FALSE,
+    original_name = c(
+      "LANDSAT/LC08/C01/T1_SR",
+      "COPERNICUS/S2_SR",
+      "NASA_USDA/HSL/SMAP_soil_moisture"
+    ),
+    pecan_code = c("l8", "s2", "smap")
+  )
+  getpecancode <- collection_lut$pecan_code
+  names(getpecancode) <- collection_lut$original_name
+  
+  if (source == "gee") {
+    collection = unname(getpecancode[collection])
+  }
+  
+  req_start <- start
+  req_end <- end
+  
+  raw_file_name = construct_raw_filename(collection, siteid_short, scale, projection, qc)
+  
+  coords = unlist(PEcAn.DB::db.query(sprintf("select ST_AsGeoJSON(geometry) from sites where id=%f", siteid), con = dbcon), use.names=FALSE)
+  
+  # check if any data is already present in the inputs table
+  existing_data <-
+    PEcAn.DB::db.query(paste0("SELECT * FROM inputs WHERE site_id=", siteid), dbcon)
+  if (nrow(existing_data) >= 1) {
```

Comment:
I wonder if this bit should be refactored to its own function, e.g. remotedata_db_check